subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractCrossVersionPerformanceTestRunner.groovy

```diff
@@ -138,92 +138,90 @@ class AbstractCrossVersionPerformanceTestRunner extends PerformanceTestSpec {
-                // for snapshots, we don't have a cheap way to check if it really exists, so we'll just
-                // blindly add it to the list and trust the test author
-                // Only active rc versions are listed in all-released-versions.properties that ReleasedVersionDistributions uses
-                addMostRecentRelease = false
-                resultBaselineVersions.add(version)
-            } else {
-                throw new RuntimeException("Cannot find Gradle release that matches version '$version'")
-            }
+        if (resolvedVersions.isEmpty() || addMostRecentRelease(overrideBaselinesProperty, versions)) {
+            // Always include the most recent final release if we're not testing against a nightly or a snapshot
+            resolvedVersions.add(releases.mostRecentRelease.version.version)
         }
 
-        if (allBaselineVersionsAreFiltered(resultBaselineVersions, versions, minimumBaseVersion)) {
-            throw new RuntimeException("All versions are filtered out by minimumBaseVersion!")
-        }
+        resolvedVersions.removeAll { !versionMeetsLowerBaseVersionRequirement(it, minimumBaseVersion) }
 
-        if (resultBaselineVersions.empty || addMostRecentRelease) {
-            // Always include the most recent final release if we're not testing against a nightly or a snapshot
-            resultBaselineVersions.add(mostRecentRelease)
+        if (resolvedVersions.isEmpty()) {
+            Assume.assumeFalse(isHistoricalPerformanceTest(versions))
         }
 
-        resultBaselineVersions
+        assert !resolvedVersions.isEmpty(): "No versions selected: ${versions}"
+
+        resolvedVersions
+    }
+
+    static boolean addMostRecentRelease(String overrideBaselinesProperty, List<String> versions) {
+        if (overrideBaselinesProperty) {
+            return false
+        }
+        return !versions.any { it == 'last' || it == 'nightly' || isRcVersionOrSnapshot(it) }
     }
 
-    static boolean allBaselineVersionsAreFiltered(LinkedHashSet<String> resultBaselineVersions, Iterable<String> versionsToBeAdded, String minimumBaseVersion) {
-        return resultBaselineVersions.empty && minimumBaseVersion != null && versionsToBeAdded.any { baseVersionLowerThanMinimunRequirement(GradleVersion.version(it), minimumBaseVersion) }
+    static boolean versionMeetsLowerBaseVersionRequirement(String targetVersion, String minimumBaseVersion) {
+        return minimumBaseVersion == null || GradleVersion.version(targetVersion).baseVersion >= GradleVersion.version(minimumBaseVersion)
     }
 
-    private static boolean baseVersionLowerThanMinimunRequirement(GradleVersion version, String minimumBaseVersion) {
-        return version.getBaseVersion() < GradleVersion.version(minimumBaseVersion)
+    static boolean isHistoricalPerformanceTest(List<String> versions) {
+        // Currently only historical performance test has more than 3 baseline versions
+        return versions.size() > 3
```

Comment:
Can't we do something better here? Adding a system property to the test task seems to be a simple way.

Reply:
Use `channel` system property instead.